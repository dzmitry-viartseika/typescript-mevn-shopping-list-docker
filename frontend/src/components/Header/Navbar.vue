<template>
  <div class="app-header">
    <div class="app-header__logo"
         @click="proceedTo('/')"
    >
      LOGO
    </div>
    <header-navigation
      :navigation-list="navigationList"
    />
  </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';
import { mixins } from 'vue-class-component';
import navigationMenuItems from '@/constants/headerNavigation';
import { INavigationMenuItem } from '@/types/INavigationMenuItem';
import HeaderNavigation from '@/components/Header/HeaderNavigation.vue';
import RouterUtils from '@/mixins/RouterUtils';

@Component({
  components: {
    HeaderNavigation,
  },
})
export default class Navbar extends mixins(RouterUtils) {
  navigationList: INavigationMenuItem[] = navigationMenuItems;
}
</script>

<style scoped lang="scss">
@import "./src/assets/scss/variables";

  .app-header {
    display: flex;
    justify-content: space-between;
    padding: 10px;
    align-items: center;
    background: $color-title;
    color: $color-white;

    &__logo {
      cursor: pointer;
    }
  }
</style>
