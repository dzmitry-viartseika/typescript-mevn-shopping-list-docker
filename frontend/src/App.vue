<template>
  <div class="app">
    <Navbar />
    <router-view/>
  </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';
import Navbar from '@/components/Header/Navbar.vue';

@Component({
  components: {
    Navbar,
  },
})
export default class App extends Vue {}
</script>

<style lang="scss">
@import "./assets/scss/style";

@font-face {
  font-family: 'Ubuntu';
  font-display: swap;
  src: url('./assets/fonts/ubuntu/ubuntu.woff') format('woff');
  font-style: normal;
}
@font-face {
  font-family: 'Ubuntu-medium';
  src: url('./assets/fonts/ubuntu/ubuntumedium.woff') format('woff');
  font-style: normal;
}
@font-face {
  font-family: 'Ubuntu-bold';
  src: url('./assets/fonts/ubuntu/ubuntubold.woff') format('woff');
  font-style: normal;
}
</style>
