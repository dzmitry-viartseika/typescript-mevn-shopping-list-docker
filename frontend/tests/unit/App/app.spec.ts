import { shallowMount, createLocalVue } from '@vue/test-utils';
import VueRouter from 'vue-router';
import App from '@/App.vue';
import Navbar from '@/components/Header/Navbar.vue';

const localVue = createLocalVue();
localVue.use(VueRouter);
const router = new VueRouter();

shallowMount(App, {
  localVue,
  router,
  stubs: ['router-link', 'router-view'],
});

describe('App.vue', () => {
  const wrapper = shallowMount(App);
  test('make App snapshot', () => {
    expect((wrapper.element)).toMatchSnapshot();
  });

  test('check rendering Navbar component', () => {
    const theNavbar = wrapper.findComponent(Navbar);
    expect(theNavbar.exists()).toBe(true);
  });
});
