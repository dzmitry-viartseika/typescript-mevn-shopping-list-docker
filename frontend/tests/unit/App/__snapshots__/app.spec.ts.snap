// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue make App snapshot 1`] = `
<div
  class="app"
>
  <navbar-stub />
   
  <router-view />
</div>
`;
